sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -907,6 +922,7 @@ primaryExpression
     | qualifiedName DOT ASTERISK                                                               #star
     | LEFT_PAREN namedExpression (COMMA namedExpression)+ RIGHT_PAREN                          #rowConstructor
     | LEFT_PAREN query RIGHT_PAREN                                                             #subqueryExpression
+    | IDENTIFIER_KW LEFT_PAREN expression RIGHT_PAREN                                          #identifierClause
```

Comment:
we might be OK with just this change in the grammar file to introduce the new IDENTIFIER clause feature, and not any of the other changes? We could always still do the other changes as well, but it could behoove us to split that into a separate PR?

Reply:
What would be the point of that?